Figgle.Generator/FiggleSourceGenerator.cs

```diff
@@ -184,6 +190,26 @@ partial class {{type.Class}}
         }
     }
 
+    private static FiggleFont? TryParseFromAdditionalFiles(
+        ImmutableDictionary<string, AdditionalText> additionalFiles,
+        string fontName)
+    {
+        if (!additionalFiles.TryGetValue(fontName, out var additionalFile))
+        {
+            return null;
+        }
+
+        var text = additionalFile.GetText();
+        if (text is null)
+        {
+            return null;
+        }
+
+        var textEncoding = text.Encoding ?? Encoding.UTF8;
+        using var stream = new MemoryStream(textEncoding.GetBytes(text.ToString()));
```

Comment:
Would be nice if Roslyn's [`SourceTextStream`](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/Text/SourceTextStream.cs) was `public`. cc @jaredpar